server/instrumentation-backend/src/main/java/sh/calaba/instrumentationbackend/actions/text/InfoMethodUtil.java

```diff
@@ -32,6 +34,13 @@ public static InputConnection getInputConnection() throws UnexpectedInputMethodM
         Context context = InstrumentationBackend.instrumentation.getTargetContext();
 
         try {
+            if (Build.VERSION.SDK_INT > 27) {
+                EditorInfo info = new EditorInfo();
+                InputConnection inputConnection = getServedView().onCreateInputConnection(info);
+
+                return inputConnection;
+            }
+
```

Comment:
Have you tested how this behaves on earlier droids?

Reply:
- really asking because it is tempting to kick out the old codepath altogether and only use onCreateInputConnection()

Reply:
I remember some issues on Android 4. I will check this again.

Reply:
Looks fine without old logic if we will keep sdk verification in other places. On Android 4 devices we have servedView instanceof WebView and correct input connection, so we don't need to use js logic on old devices (especially because JS logic use modern things like a new way to trigger events which I really like).